rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -605,11 +605,11 @@ SecRule REQUEST_COOKIES|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i)1\
 # (consult https://coreruleset.org/docs/development/regex_assembly/ for details):
 #   crs-toolchain regex update 942550
 #
-SecRule REQUEST_FILENAME|REQUEST_COOKIES|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx [\"'`][\[\{].*[\]\}][\"'`].*(::.*jsonb?)?.*(?:(?:@|->?)>|<@|\?[&\|]?|#>>?|[<>]|<-)|(?:(?:@|->?)>|<@|\?[&\|]?|#>>?|[<>]|<-)[\"'`][\[\{].*[\]\}][\"'`]|json_extract.*\(.*\)" \
+SecRule REQUEST_FILENAME|REQUEST_COOKIES|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i)(?:::[\s\x0b]*?(/\*.*?\*/)?[\s\x0b]*?jsonb?)?[\s\x0b]*?(?:(?:@|->?)>|<@|\?[&\|]?|#>>?|[<>]|<-)|(?:(?:@|->?)>|<@|\?[&\|]?|#>>?|[<>]|<-)?[\"'`][\[\{][^#\]\}]*[\]\}]+[\"'`]|\bjson_extract\b[^\(]*\([^\)]*\)" \
     "id:942550,\
     phase:2,\
     block,\
-    t:none,t:urlDecodeUni,t:lowercase,t:removeWhitespace,\
+    t:none,t:urlDecodeUni,t:removeWhitespace,\
```

Comment:
You're matching whitespace in the regex. If you need that, remove `removeWhitespace`, otherwise, remove the whitespace matches from the regex.

Reply:
Removed the whitespace matches from the regex. Now we rely on the transformation.